amsclient/amsclient.go

```diff
@@ -0,0 +1,131 @@
+// Copyright 2021 Red Hat, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package amsclient
+
+import (
+	"fmt"
+	"strings"
+
+	sdk "github.com/openshift-online/ocm-sdk-go"
+	"github.com/rs/zerolog/log"
+
+	"github.com/RedHatInsights/insights-results-aggregator/types"
+)
+
+// AMSClient allow us to interact the AMS API
+type AMSClient struct {
+	connection *sdk.Connection
+	pageSize   int
+}
+
+// NewAMSClient creates an AMSClient from the configuration
+func NewAMSClient(conf Configuration) (*AMSClient, error) {
+	conn, err := sdk.NewConnectionBuilder().
+		URL(conf.URL).
+		Tokens(conf.Token).
+		Build()
+
+	if err != nil {
+		log.Error().Err(err).Msg("Unable to build the connection to AMS API")
+		return nil, err
+	}
+
+	var pageSize int
+	if conf.PageSize == 0 {
+		pageSize = 100
+	} else {
+		pageSize = conf.PageSize
+	}
+
+	return &AMSClient{
+		connection: conn,
+		pageSize:   pageSize,
+	}, nil
+}
+
+// GetClustersForOrganization retrieves the clusters for a given organization using the default client
+// it allows to filter the clusters by their status (statusNegativeFilter will exclude the clusters with status in that list)
+func (c *AMSClient) GetClustersForOrganization(orgID types.OrgID, statusFilter, statusNegativeFilter []string) []types.ClusterName {
+	var retval []types.ClusterName = []types.ClusterName{}
+
+	internalOrgID, err := c.getInternalOrgIDFromExternal(orgID)
+	if err != nil {
+		return retval
+	}
+
+	searchQuery := fmt.Sprintf("organization_id is '%s'", internalOrgID)
+
+	if len(statusFilter) > 0 {
```

Comment:
I'd refactor it into a new function

Reply:
Agree, moving the filter string generation to an utility function